pkg/controller/policyrecommendation_controller.go

```diff
@@ -18,59 +18,194 @@ package controller
+			Name:      policyreco.Name,
+			Namespace: policyreco.Namespace,
+		},
+		Spec: v1alpha1.PolicyRecommendationSpec{
+			QueuedForExecution:      &FALSE_BOOL,
+			TargetHPAConfiguration:  *targetHPAReco,
+			Policy:                  policyName,
+			CurrentHPAConfiguration: *hpaConfigToBeApplied,
+			TransitionedAt:          &transitionedAt,
+			GeneratedAt:             &now,
+		},
+	}
+	logger.V(0).Info("Policy Patch", "PolicyReco", *policyRecoPatch)
+	if err := r.Patch(ctx, policyRecoPatch, client.Apply, client.ForceOwnership, client.FieldOwner(POLICY_RECO_WORKFLOW_CTRL_NAME)); err != nil {
+		logger.Error(err, "Error patching the policy reco object")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	logger.V(0).Info("Policy Patch", "PolicyReco", *policyRecoPatch)
+
+	statusPatch := &v1alpha1.PolicyRecommendation{
+		TypeMeta: policyreco.TypeMeta,
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      policyreco.Name,
+			Namespace: policyreco.Namespace,
+		},
+		Status: v1alpha1.PolicyRecommendationStatus{},
+	}
+	if err := r.Status().Patch(ctx, statusPatch, client.Apply, getSubresourcePatchOptions()); err != nil {
+		logger.Error(err, "Failed to patch the status")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	r.Recorder.Event(&policyreco, EVENT_TYPE_NORMAL, "HPARecommendationGenerated", fmt.Sprintf("The HPA recommendation has been generated successfully. The current policy this workload is at %s.", policyName))
+	logger.V(1).Info("Successfully generated HPA Recommendation.")
 	return ctrl.Result{}, nil
 }
 
+func retrieveTransitionTime(hpaConfigToBeApplied *v1alpha1.HPAConfiguration, policyreco v1alpha1.PolicyRecommendation) metav1.Time {
+	if !hpaConfigToBeApplied.DeepEquals(policyreco.Spec.CurrentHPAConfiguration) {
+		return metav1.Now()
+	}
+	return *policyreco.Spec.TransitionedAt
+}
+
+func getSubresourcePatchOptions() *client.SubResourcePatchOptions {
+	patchOpts := client.PatchOptions{}
+	client.ForceOwnership.ApplyToPatch(&patchOpts)
+	client.FieldOwner(POLICY_RECO_WORKFLOW_CTRL_NAME).ApplyToPatch(&patchOpts)
+	return &client.SubResourcePatchOptions{
+		PatchOptions: patchOpts,
+	}
+}
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *PolicyRecommendationReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// predicates to filter events with updates to QueuedForExecution or QueuedForExecutionAt
+	queuedTaskPredicate := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			objSpec := e.Object.(*v1alpha1.PolicyRecommendation).Spec
+			switch {
+			case *objSpec.QueuedForExecution == true:
+				return true
+			default:
+				return false
+			}
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			oldObjSpec := e.ObjectOld.(*v1alpha1.PolicyRecommendation).Spec
+			newObjSpec := e.ObjectNew.(*v1alpha1.PolicyRecommendation).Spec
+			// If it hasn't been touched by the registrar don't reconcile
```

Comment:
PolicyRecommendation object is created by the registrar only. In which scenario would this condition be true?